kompari-tasks/src/task.rs

```diff
@@ -69,6 +70,11 @@ impl Task {
                     ds_args,
                 )?;
             }
+            Command::SizeCheck(sc_args) => {
+                let results =
+                    check_size_optimizations(self.diff_config.left_path(), sc_args.optimize)?;
```

Comment:
There probably should be a way for this to give an error code, if a more than (say) 40% improvement is possible. We'd then check this in CI.

Incidentally, we also want to limit the size of a single snapshot, again in CI. That would also make sense at the same time, or using the same intermediate value.

These can both be follow-ups